### ISSUE TITLE
`tcomb-validation` error message doesn't contain struct name

### ISSUE THREAD

**Issue by ArmorDarks** [gcanti/tcomb-validation#56]
For some reason `tcomb-validation` outputs different from `tcomb` error message — it doesn't account for `name` option of `t.struct`.

Example:

``` js
const Schema = t.struct({
  name: t.String,
  surname: t.maybe(t.String),
  age: t.Integer,
  tags: t.list(t.String)
}, { name: 'Person', strict: true })

Schema(data)
console.log(tv.validate(data, Schema).errors)
```

`tcomb` will produce

```
[tcomb] Invalid value undefined supplied to Person/name: String
```

`tcomb-validation` will produce

```
Invalid value undefined supplied to /name: String
```

Note how `tcomb` prints `Person` as name from `struct`, but `tcomb-validate` does not.

Is it per design?

**Comment by gcanti** [comment]
> Is it per design?

Yes (in order to handle nested types), however you can change the default behaviour adding a prefix through the `path` option

```js
function myvalidate(value, type) {
  return t.validate(value, type, { path: [type.displayName] })
}
```

**Comment by ArmorDarks** [comment]
Hm, I see. Though, it's unclear, why `tcomb-validate` can't try to access struct's name at first place, without need to specify path?

Also, maybe it worth to add in readme?

Obviously, I'm not familiar with all pitfalls and `tcomb` is very new for me, so, I'm sorry if I'm raising something that already has been discussed hundreds times.

**Comment by gcanti** [comment]
IIRC validations are assumed to be *relative* (hence the `"/name"` string in the error message) to a root type which is never mentioned, except in this trivial case

```js
console.log(t.validate(1, Schema).errors) // => Invalid value 1 supplied to Person
```

The path of the root however can be customized through the `path` option.

(I don't necessarily agree with the old me on this topic, I'm just explaining why it works like that)

**Comment by ArmorDarks** [comment]
Hm, I'm quite confused. I'm probably missing something...

So, this

```js
console.log(t.validate(1, Schema).errors) // => Invalid value 1 supplied to Person
```

will mention `Schema` name (`Person`)?

But this one will not?

```js
console.log(tv.validate(data, Schema).errors) // => Invalid value undefined supplied to /name: String
```
